### PR TITLE
Adding attachment fields to describe the environment in eBPF ALS

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

File: ebpf/accesslog.proto [hunk 1]
```diff
@@ -80,6 +80,21 @@ message AccessLogConnection {
...
+    // detect contains the environment information
+    oneof environment {
+        // detect the connection have a zTunnel environment(ambient istio)
+        ZTunnelAttachmentEnvironment zTunnel = 1;
+    }
+}
+
+message ZTunnelAttachmentEnvironment {
+    // the real destination ip address of the connection
+    string real_destination_ip = 1;
```
Comment (wu-sheng) [inline ebpf/accesslog.proto:97]: Could we add an emun `by` to indicate how we resolve the real IP?

Comment (mrproliu) [reply ebpf/accesslog.proto:97]: Where do we need to use this enum?

Comment (wu-sheng) [reply ebpf/accesslog.proto:97]: We don't, but we can tell which part of the logic made this. Helpful for debug only.

Comment (mrproliu) [reply ebpf/accesslog.proto:97]: I have added a new enum to describe how the rover detects and resolves the real dest IP. Is this work for you?

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**